crates/noirc_evaluator/src/ssa/node.rs

```diff
@@ -178,12 +191,16 @@ pub enum ObjectType {
     Unsigned(u32), //bit size
     Signed(u32),   //bit size
     Pointer(ArrayId),
-    //custom(u32),   //user-defined struct, u32 refers to the id of the type in...?todo
+
+    Function(ArrayIdSet), // Function with a set of arrays that may be returned
```

Comment:
array id should not be part of the type.
Normally a function type would be its signature, but I am not sure how useful it will be.

Reply:
I've pushed commit ee0c4e7 removing this tracking in favor of creating a fresh array at each callsite. This is certainly suboptimal since we must decide on a size for the array and if the user indexes over this size the compiler crashes in anchor.rs. Currently I've set this maximum size to 100_000.
We can track the maximum possible array size returned by the functions used, but then we'd need to bring back something similar to the ArrayIdSet ids but for array sizes instead.